apps/src/blockly/addons/functionEditor.js

```diff
@@ -203,7 +207,9 @@ export default class FunctionEditor {
 
     const isBehavior = type === BLOCK_TYPES.behaviorDefinition;
     // We do not want to show the delete button for behaviors that are not user-created
-    const hideDeleteButton = isBehavior && !this.block.userCreated;
+    // or are in read only workspaces.
+    const hideDeleteButton =
+      this.isReadOnly || (isBehavior && !this.block.userCreated);
```

Comment:
Just calling out that my PR also touches this behavior, and this will eventually be (assuming there aren't changes to my PR):
```
this.isReadOnly || !this.block.userCreated
```
Relying on userCreated exclusively is parity with CDO.